### DISCUSSION TITLE
Optimistic delete with infinite (i.e. paginated) query?

### DISCUSSION THREAD

**Discussion by patrick-r-son** [TanStack/query#7789] [Q&A] [answered]
I have a UI consisting of an infinite list of items, and the user can delete items with a mutation. What kind of considerations need to be made when doing optimistic removals to an infinite paginated query?  Does anyone have a real-life-applicable example?

**Comment by TkDodo** [comment]
with optimistic updates, you ideally do exactly what the server will also do.

**Reply by patrick-r-son** [reply]
Well yeah, but with the paged response, I just delete that item from the cache, now that page only has 9 items while the page size is supposed to be 10. I was just wondering if that would cause any unforeseen weirdness with the way `useInfiniteQuery` works.

**Reply by TkDodo** [reply, marked as answer]
you can shift items around from the follow-up pages, but then the last page will only have 9 items until you refetch. I don't think it matters much which page is the one where one item is missing

**Reply by patrick-r-son** [reply]
> you can shift items around from the follow-up pages

Well in my case the items are shown in the UI as an infinitely scrolling list so I just removed the items from each page as needed and it ended up working fine.

Good point about paginated UIs though - something for others to keep in mind if they aren't doing infinite scrolling.
